package MENTORING.Mentoring;

public class Exceptions {
    public static void main(String[] args) {

//    ArithmeticException
//        try {
//        System.out.println(5/0);
//        }catch (ArithmeticException e){
//            System.out.println("sorry");

//    It is thrown when an exceptional condition has occurred in an arithmetic operation.

//    ArrayIndexOutOfBoundsException
//        int[]array = new int[4];
        //0 1 2 3 indexes
//        array[4]=4;
//    It is thrown to indicate that an array has been accessed with an illegal index. The index is either negative or greater than or equal to the size of the array.

//    RuntimeException
//        try{
//            System.out.println(2/0);
//        }catch (RuntimeException e){
//            System.out.println("sorry");
//        }
//    This represents any exception which occurs during runtime.

//    NullPointerException
//        String a = null;
//        System.out.println(a.charAt(0));
//        ArrayList<Integer> list = null;
//        System.out.println(list.get(0));
    }

//    This exception is raised when referring to the members of a null object. Null represents nothing

//    StringIndexOutOfBoundsException
//        String str = "Yusuf";
//        System.out.println(str.charAt(6));
//    It is thrown by String class methods to indicate that an index is either negative than the size of the string

//    NumberFormatException
//        String strNumber = "10";
//        String fakeNumber = "TechnoStudy";
//        int a = Integer.parseInt(strNumber);
//        System.out.println(a);
//        int b = Integer.parseInt(fakeNumber);
//        System.out.println(b);
//    This exception is raised when a method could not convert a string into a numeric format.


}